module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.hpp

```diff
@@ -18,24 +18,52 @@ namespace gui
       public:
         PhoneLockedWindow(app::ApplicationCommon *app,
                           const std::string &name,
-                          std::unique_ptr<WallpaperPresenter> &&presenter);
+                          std::unique_ptr<WallpaperPresenter> &&presenter,
+                          std::uint8_t lockScreenDeepRefreshRate);
 
         bool onInput(const InputEvent &inputEvent) override;
         void onBeforeShow(ShowMode mode, SwitchData *data) override;
         void onClose(CloseReason reason) override;
         void buildInterface() override;
         status_bar::Configuration configureStatusBar(status_bar::Configuration appConfiguration) override;
 
-        bool updateTime() override;
+        RefreshModes updateTime() override;
 
       private:
         bool processLongReleaseEvent(const InputEvent &inputEvent);
-        static constexpr auto refreshTimerName = "PhoneLockRefreshTimer";
-        static constexpr auto refreshTimeout   = std::chrono::hours(1);
-        sys::TimerHandle screenRefreshTimer;
 
         std::unique_ptr<WallpaperPresenter> wallpaperPresenter;
         bool refreshedOnPhoneLockTimeLock = false;
+
+        void initializeDeepRefreshCounter(std::uint8_t deepRefreshRate);
+
+        struct
+        {
+            std::uint8_t currentValue;
+            std::uint8_t referenceValue;
```

Comment:
why not plain `unsigned int` ? It's not like we care about integer precision here ;) 

small nitpick - do whatever you want with this comment :bowing_man:

Reply:
Just a habit from coding for smaller, 8-bit CPUs :P Changed to unsigned.